server/models/User.js

```diff
@@ -1,49 +1,53 @@
 import mongoose from "mongoose";
 const Schema = mongoose.Schema
 
-const courseCompletedSchema = new Schema({
-  courseId: {
-    type: Schema.Types.ObjectId,
-    ref: "Course",
-    required: true
-  },
-  completedLectures: [{
-    type: Schema.Types.ObjectId,
-    ref: "Lecture",
-  }]
+const courseCompletedSchema = new Schema({         
+    courseId: {   
+        type: Schema.Types.ObjectId,
+        ref: "Course",
+        required: true
+    },
+    completedLectures: [{
+        type: Schema.Types.ObjectId,
+        ref: "Lecture",
+    }]
 })
 
 const UserSchema = new Schema({
-  first_name: {
-    type: String,
-    lowercase: true,
-  },
-  last_name: {
-    type: String,
-    lowercase: true,
-  },
-  email: {
-    type: String,
-    lowercase: true,
-    trim: true,
-    required: true
-  },
-  password: {
-    type: String,
-    required: true
-  },
-  wallet_address: [{
-    type: String
-  }],
-  role: {
-    type: String,
-    required: true
-  },
-  courses: [{
-    type: courseCompletedSchema
-  }]
+    first_name: {
+        type: String,
+        lowercase: true,
+    },
+    last_name: {
+        type: String,
+        lowercase: true,
+    },
+    email: {
+        type: String,
+        lowercase: true,
+        trim: true,
+        required: true
+    },
+    password: {
+        type: String,
+        required: false
+    },
+    wallet_address: [{
+        type: String
+    }],
+    role: {
+        type: String,
+        required: true
+    },
+    source: {
+        type: String,
+        required: false
+    },
+    courses: [{
+        type: courseCompletedSchema
+    }]
```

Comment:
We don't have this atrribute anymore in the users anymore. We have a seperate model named UserCourses. Please check: https://github.com/Lumos-metaverse/education-portal-lumos/blob/staging/models/UserCourses.js